src/audio/volume_hifi3.c

```diff
@@ -60,7 +60,7 @@ static void vol_s16_to_s16(struct comp_dev *dev, struct comp_buffer *sink,
 	ae_f16x4 in_sample;
 	size_t channel;
 	int i;
-	int limit = source->size / (dev->params.channels << 1);
+	int limit = cd->source_period_bytes / (dev->params.channels << 1);
```

Comment:
is this really equivalent to the generic code?
The generic code iterations are controlled by dev->frames, here you are making a reference to the source period, is this matching?

Reply:
Yes, it's exactly the same as the generic code.
Previously it was incorrect, because it was using the size of the whole buffer instead of the size of currently processed period.

Reply:
What I was trying to say is that the code would be "more similar" if you used the same dev->frames in the loop. the source_period_bytes is derived from dev->frames, so might as well use the latter for symmetry.